### PR TITLE
[ansible-test] use newer container images

### PR DESCRIPTION
**Pull request by relrod**
##### SUMMARY
Change:
- Bump default, ansible-base, distro containers

Test Plan:
- CI, hopefully

Signed-off-by: Rick Elrod <rick@elrod.me>

##### ISSUE TYPE

- Test Pull Request

##### COMPONENT NAME

ansible-test

### CODE REVIEW

File: test/lib/ansible_test/_data/completion/docker.txt [hunk 1]
```diff
@@ -1,13 +1,13 @@
-default name=quay.io/ansible/default-test-container:2.7.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=collection
-default name=quay.io/ansible/ansible-base-test-container:1.6.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=ansible-base
+default name=quay.io/ansible/default-test-container:2.9.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=collection
+default name=quay.io/ansible/ansible-base-test-container:1.7.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=ansible-base
 alpine3 name=quay.io/ansible/alpine3-test-container:1.19.0 python=3.8
 centos6 name=quay.io/ansible/centos6-test-container:1.17.0 python=2.6 seccomp=unconfined
 centos7 name=quay.io/ansible/centos7-test-container:1.17.0 python=2.7 seccomp=unconfined
-centos8 name=quay.io/ansible/centos8-test-container:1.17.0 python=3.6 seccomp=unconfined
+centos8 name=quay.io/ansible/centos8-test-container:1.20.0 python=3.6 seccomp=unconfined
```
Comment (mattclay) [inline test/lib/ansible_test/_data/completion/docker.txt]: We might as well make this and the other version 20 containers use 21 since that's the latest tag.

**Review by mattclay — APPROVED** [review]

File: test/integration/targets/incidental_setup_mysql_db/vars/Suse.yml [hunk 2]
```diff
@@ -4,3 +4,9 @@ mysql_packages:
     - mariadb
     - python-PyMySQL
     - bzip2
+
+# opensuse15py2 brings in a bunch of py3 stuff. Nuke it.
+mysql_cleanup_packages:
+    - python3-base
+    - python3-mysqlclient
+    - libpython3_6m1_0
```
Comment (mattclay) [inline test/integration/targets/incidental_setup_mysql_db/vars/Suse.yml]: There may be collections which will need this same change to their tests.

Comment (sivel) [reply test/integration/targets/incidental_setup_mysql_db/vars/Suse.yml]: fwiw, I'm in the process of removing this incidental.

**Comment by mkrizek** [comment]
!waffling core_review

**Comment by relrod** [comment]
@mattclay to re-review

I pulled out the f33 stuff for now into its own branch since we're blocked on AZP for it. I think we can merge this in the meantime though.

**Comment by samdoran** [comment]
!waffling core_review

File: test/lib/ansible_test/_data/completion/docker.txt [hunk 3]
```diff
@@ -1,13 +1,13 @@
-default name=quay.io/ansible/default-test-container:2.7.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=collection
-default name=quay.io/ansible/ansible-base-test-container:1.6.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=ansible-base
+default name=quay.io/ansible/default-test-container:2.9.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=collection
+default name=quay.io/ansible/ansible-base-test-container:1.7.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=ansible-base
```
Comment (mattclay) [inline test/lib/ansible_test/_data/completion/docker.txt:2]: A changelog entry is needed for these updates.

Also, I'd backport them to stable-2.10. The other chagnes are probably OK to backport as well, but if not, they should be split into a separate PR.

**Review by mattclay — CHANGES_REQUESTED** [review]

**Comment by samdoran** [comment]
!waffling needs_revision

File: test/lib/ansible_test/_data/completion/docker.txt (hunk 3 above)
Comment (relrod) [reply test/lib/ansible_test/_data/completion/docker.txt:2]: Updated changelog entry to call out those two containers. And I was planning to backport everything to 2.10.

Comment (mattclay) [reply test/lib/ansible_test/_data/completion/docker.txt:2]: The default containers also move from Python 3.9 RC1 to GA, so that should be included in the changelog.

**Review by mattclay — APPROVED** [review]
